### PR TITLE
Add/update IOS XR information

### PR DESCRIPTION
**Pull request by glennmatthews**
Fill XR provider and property support and caveats in README.md. This also transposes the property 'OS and Module minimum version by platform' tables to make it easier to add new platforms without creating nasty merge conflicts - discussed this plan with @chrisvanheuveln this morning.

### CODE REVIEW

**Comment by glennmatthews** [comment]
@helenhzhang Can you take a look at the updated bgp_neighbor_af caveat table and see if I missed anything?

**Comment by HelenHZhang** [comment]
I looked at bgp_neighbor_af caveat table. It's correct.

File: README.md [hunk 1]
```diff
@@ -596,16 +641,45 @@ Address Family Identifier (AFI). Required. Valid values are ipv4 and ipv6.
...
 
 #### <a name="cisco_bgp-caveats">Caveats</a>
 
 | Property | Caveat Description |
 |:--------|:-------------|
-| `disable_policy_batching_ipv4` <br> `disable_policy_batching_ipv4` <br> `neighbor_down_fib_accelerate` <br> `reconnect_interval` | Not supported on N56xx, N6k, N7k |
+| `bestpath_med_confed` | Only supported in global BGP context in IOS XR |
+| `bestpath_med_non_deterministic` | Not supported on IOS XR |
+| `cluster_id` | Only supported in global BGP context in IOS XR |
+| `confederation_id` | Only supported in global BGP context in IOS XR |
```
Comment (chburket) [inline README.md]: I believe confederation_peers is also only supported in global BGP for XR.

File: README.md [hunk 2]
```diff
@@ -742,16 +816,36 @@ Set bgp hold timer. Valid values are Integer, keyword 'default'.
...
 #### <a name="cisco_bgp_af-caveats">Caveats</a>
 
 | Property | Caveat Description |
 |:--------|:-------------|
-| `advertise_l2vpn_evpn ` | Not supported on N30xx, N31xx, N6k |
+| `additional_paths_install` | Not supported on IOS XR |
+| `advertise_l2vpn_evpn` | Not supported on N30xx, N31xx, N6k, IOS XR |
+| `client_to_client` | Only supported in global BGP context in IOS XR |
+| `dampen_igp_metric` | Not supported on IOS XR |
+| `dampening_state` | Only supported in global BGP context in IOS XR |
```
Comment (chburket) [inline README.md]: Any reason why this property is not in alpha order?

File: README.md (hunk 1 above)
Comment (glennmatthews) [inline README.md]: Huh, interesting. The YAML for this property explicitly includes a VRF option, but test_router_bgp.rb minitest does indeed expect XR to fail in a VRF for this property. I was misled by the YAML. :-)

File: README.md (hunk 2 above)
Comment (glennmatthews) [inline README.md]: Good catch. Will fix.

File: README.md (hunk 1 above)
Comment (chburket) [inline README.md]: Should we fix the YAML?

Comment (glennmatthews) [inline README.md]: Yup - looking into it. :)

Comment (glennmatthews) [inline README.md]: cisco/cisco-network-node-utils@dbecda15acffc9458eaabde3e606942c75b48a8e

File: README.md [hunk 3]
```diff
@@ -471,10 +491,15 @@ Enable/disable mschapv2 for AAA Authentication Login.
...
-| OS Image | 7.0(3)I2(1) | 7.0(3)I2(1) | 7.0(3)I2(1) | 7.3(0)N1(1) | 7.3(0)N1(1) | 7.3(0)D1(1) |
-| Puppet Module | 1.2.0 | 1.2.0 | 1.2.0 | unsupported | unsupported | unsupported |
+| Platform | OS Minimum Version | Module Minimum Version |
+|----------|:------------------:|:----------------------:|
+| N9k      | 7.0(3)I2(1)        | 1.2.0                  |
+| N30xx    | 7.0(3)I2(1)        | 1.2.0                  |
+| N31xx    | 7.0(3)I2(1)        | 1.2.0                  |
+| N56xx    | 7.3(0)N1(1)        | unsupported            |
+| N6k      | 7.3(0)N1(1)        | unsupported            |
+| N7k      | 7.3(0)D1(1)        | unsupported            |
```
Comment (chrisvanheuveln) [inline README.md]: We talked about leaving the OS min versions blank for unsupported properties. Still want to change that?

File: README.md [hunk 4]
```diff
@@ -413,10 +423,15 @@ The following resources are listed alphabetically.
...
-| OS Image | 7.0(3)I2(1) | 7.0(3)I2(1) | 7.0(3)I2(1) | 7.3(0)N1(1) | 7.3(0)N1(1) | 7.3(0)D1(1) |
-| Puppet Module | 1.0.1 | 1.0.1 | 1.0.1 | 1.2.0 | 1.2.0 | 1.2.0 |
+| Platform | OS Minimum Version | Module Minimum Version |
+|----------|:------------------:|:----------------------:|
+| N9k      | 7.0(3)I2(1)        | 1.0.1                  |
+| N30xx    | 7.0(3)I2(1)        | 1.0.1                  |
+| N31xx    | 7.0(3)I2(1)        | 1.0.1                  |
+| N56xx    | 7.3(0)N1(1)        | 1.2.0                  |
+| N6k      | 7.3(0)N1(1)        | 1.2.0                  |
+| N7k      | 7.3(0)D1(1)        | 1.2.0                  |
```
Comment (chrisvanheuveln) [inline README.md]: We talked about leaving the OS min versions blank for unsupported properties. Still want to change that?
 Add a line note

File: README.md [hunk 5]
```diff
@@ -441,10 +456,15 @@ This provider allows raw configurations to be managed by Puppet. It serves as a
...
-| OS Image | 7.0(3)I2(1) | 7.0(3)I2(1) | 7.0(3)I2(1) | 7.3(0)N1(1) | 7.3(0)N1(1) | 7.3(0)D1(1) |
-| Puppet Module | 1.2.0 | 1.2.0 | 1.2.0 | unsupported | unsupported | unsupported |
+| Platform | OS Minimum Version | Module Minimum Version |
+|----------|:------------------:|:----------------------:|
+| N9k      | 7.0(3)I2(1)        | 1.2.0                  |
+| N30xx    | 7.0(3)I2(1)        | 1.2.0                  |
+| N31xx    | 7.0(3)I2(1)        | 1.2.0                  |
+| N56xx    | 7.3(0)N1(1)        | unsupported            |
+| N6k      | 7.3(0)N1(1)        | unsupported            |
+| N7k      | 7.3(0)D1(1)        | unsupported            |
```
Comment (chrisvanheuveln) [inline README.md]: We talked about leaving the OS min versions blank for unsupported properties. Still want to change that?

Comment (glennmatthews) [inline README.md]: Yeah, probably so. Let me see what I can do.

Comment (glennmatthews) [inline README.md]: Changed the min version to 'unsupported' rather than blank.

**Pull request merged**